actors/builtin/miner/miner_state.go

```diff
@@ -259,6 +273,184 @@ func (st *State) ComputeProvingSet(store adt.Store) ([]abi.SectorInfo, error) {
 	return sectorInfos, nil
 }
 
+func (st *State) addPledge(store adt.Store, currEpoch abi.ChainEpoch, pledgeAmount abi.TokenAmount) error {
+	// Assert(pledgeAmount.GreaterThanEqual(big.Zero()))
+
+	// pledge := abi.LockedFund{
+	// 	CliffStartEpoch: currEpoch + PledgeCliff,
+	// 	FullyVestedEpoch: currEpoch + PledgeCliff + PledgeVestingPeriod,
+	// 	Value: pledgeAmount,
+	// 	AmountWithdrawn: abi.NewTokenAmount(0),
+	// 	AmountSlashed: abi.NewTokenAmount(0),
+	// 	VestingFunction: abi.Linear,
+	// }
+
+	// var pc PledgeCollateral
+	// store.Get(st.PledgeCollateral, &pc)
+
+	// pledges := adt.AsMultimap(store, pc.LockedFunds)
+	// lockedFunds, found, err := pledges.Get(cliffStartEpochKey(pledge.CliffStartEpoch))
+	// if err != nil{
+	// 	return errors.Wrapf(err, "failed to add pledge: %v", err)
+	// }
+
+	// if !found {
+	// 	lockedFunds, err = adt.MakeEmptyArray(store)
+	// 	if err != nil {
+	// 		return errors.Wrapf(err, "failed to initialize in add pledge: %v", err)
+	// 	}
+	// }
+
+	// lockedFunds.AppendContinuous(pledge)
+	// // update the bitfield
+	// pc.LockedIndices.Set(uint64(pledge.CliffStartEpoch)) // TODO: ChainEpoch should be uint64
+	// pc.LockedFunds = lockedFunds.Root()
```

Comment:
I think `lockedFunds` is to be inserted or updated within `pledges`; it is `pledges.Root()` which then needs to be updated into `pc` here.